internal/commands/vaultsecrets/run/run.go

```diff
@@ -0,0 +1,186 @@
+	Profile *profile.Profile
+	IO      iostreams.IOStreams
+	Output  *format.Outputter
+	Logger  hclog.Logger
+
+	AppName       string
+	Command       []string
+	PreviewClient preview_secret_service.ClientService
+	Client        secret_service.ClientService
+}
+
+func NewCmdRun(ctx *cmd.Context, runF func(*RunOpts) error) *cmd.Command {
+	opts := &RunOpts{
+		Ctx:           ctx.ShutdownCtx,
+		Profile:       ctx.Profile,
+		IO:            ctx.IO,
+		Output:        ctx.Output,
+		PreviewClient: preview_secret_service.New(ctx.HCP, nil),
+		Client:        secret_service.New(ctx.HCP, nil),
+	}
+
+	cmd := &cmd.Command{
+		Name:      "run",
+		ShortHelp: "Run a process with secrets from a Vault Secrets app.",
+		LongHelp: heredoc.New(ctx.IO).Must(`
+		The {{ template "mdCodeOrBold" "hcp vault-secrets run" }} command lets you run 
+		the provided command as a child process while injecting all of the app's secrets
+		as environment variables, with all secret names converted to upper-case. The stdout
+		and stderr from the child process are forwarded to the top level {{ template "mdCodeOrBold" "hcp vault-secrets run" }} command.
+		`),
+		Examples: []cmd.Example{
+			{
+				Preamble: `Inject secrets as environment variables:`,
+				Command: heredoc.New(ctx.IO, heredoc.WithPreserveNewlines()).Must(`
+				$ hcp vault-secrets run 'env'
+				`),
+			},
+		},
+		Args: cmd.PositionalArguments{
+			Args: []cmd.PositionalArgument{
+				{
+					Name:          "COMMAND",
+					Documentation: "Defines the invocation of the child process to inject secrets to.",
+				},
+			},
+			Validate: cmd.ArbitraryArgs,
+		},
+		Flags: cmd.Flags{
+			Local: []*cmd.Flag{
+				{
+					Name:         "app",
+					DisplayValue: "NAME",
+					Description:  "The application you want to pull all secrets from.",
+					Value:        appname.Flag(),
+				},
+			},
+		},
+		PersistentPreRun: func(c *cmd.Command, args []string) error {
+			return appname.Require(ctx)
+		},
+		RunF: func(c *cmd.Command, args []string) error {
+			opts.Command = args[0:]
+			opts.AppName = appname.Get()
+
+			if runF != nil {
+				return runF(opts)
+			}
+			return runRun(opts)
+		},
+	}
+	for _, f := range cmd.Flags.Local {
+		if f.Name == "app" {
+			f.Autocomplete = helper.PredictAppName(ctx, cmd, opts.PreviewClient)
+		}
+	}
+
+	return cmd
+}
+
+func runRun(opts *RunOpts) (err error) {
+	if len(opts.Command) == 0 {
+		return fmt.Errorf("failed to run app secrets from name %q - no command provided", opts.AppName)
+	}
+
+	envSecrets, err := getAllSecretsForEnv(opts)
+	if err != nil {
+		return fmt.Errorf("failed to run with secrets in app %q: %w", opts.AppName, err)
+	}
+
+	childProcess := setupChildProcess(opts.Ctx, opts.Command, envSecrets)
+
+	err = childProcess.Run()
+	if err != nil {
+		var exitErr *exec.ExitError
+		if errors.As(err, &exitErr) {
+			exitCode := exitErr.ExitCode()
+			return fmt.Errorf("failed to run with secrets in app %q; received code %v : %w", opts.AppName, exitCode, err)
```

Comment:
I just made this PR for you: https://github.com/hashicorp/hcp/pull/103. Would be good to use that so that the exit code gets passed through to the user.